### PR TITLE
Pod labels for Kiali Operator Helm Chart

### PR DESCRIPTION
**Pull request by magdalenakuhn**
https://github.com/kiali/kiali/issues/5832

### CODE REVIEW

File: kiali-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ spec:
         # required for the operator SDK metric service selector
         name: {{ include "kiali-operator.fullname" . }}
       {{- include "kiali-operator.labels" . | nindent 8 }}
+      {{- include "kiali-server.labels" . | nindent 8 }}
```
Comment (jmazzitelli) [inline kiali-operator/templates/deployment.yaml:22]: this should not be here. the operator is unrelated to the kiali-server (in fact, I don't even think kiali-server.labels is even defined in the operator helm chart).

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

File: kiali-operator/templates/deployment.yaml (hunk 1 above)
Comment (magdalenakuhn) [inline kiali-operator/templates/deployment.yaml:22]:
```suggestion
```

Comment (magdalenakuhn) [reply kiali-operator/templates/deployment.yaml]: Fixed in 901bfdc2165055f7e2b4cbf731cf29ec12ce2e11. Sorry, don't know how this slipped through.

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
CI build failed. check the errors and correct.

**Comment by jmazzitelli** [comment]
Did you test this? Because it doesn't even build.

```
$ make build-helm-charts 
Will use this helm executable: /home/jmazzite/source/kiali/helm-charts/_output/helm-install/helm
Building Helm Chart for Kiali operator
==> Linting /home/jmazzite/source/kiali/helm-charts/_output/charts/kiali-operator
[ERROR] templates/: template: kiali-operator/templates/deployment.yaml:22:20: executing "kiali-operator/templates/deployment.yaml" at <.Values.deployment.podLabels>: nil pointer evaluating interface {}.podLabels

Error: 1 chart(s) linted, 1 chart(s) failed
make: *** [Makefile:75: .build-helm-chart-operator] Error 1
```

**Comment by magdalenakuhn** [comment]
> Did you test this? Because it doesn't even build.
> 
> ```
> $ make build-helm-charts 
> Will use this helm executable: /home/jmazzite/source/kiali/helm-charts/_output/helm-install/helm
> Building Helm Chart for Kiali operator
> ==> Linting /home/jmazzite/source/kiali/helm-charts/_output/charts/kiali-operator
> [ERROR] templates/: template: kiali-operator/templates/deployment.yaml:22:20: executing "kiali-operator/templates/deployment.yaml" at <.Values.deployment.podLabels>: nil pointer evaluating interface {}.podLabels
> 
> Error: 1 chart(s) linted, 1 chart(s) failed
> make: *** [Makefile:75: .build-helm-chart-operator] Error 1
> ```

make build-helm-charts now runs successfully on my machine! Fixed in 7e8767b84ffdafb9ba1c422a3fe86b194328235c. Maybe after this PR I can open a PR to fix the different conventions between server and operator ( values.deployment.pod_labels and values.podLabels, value.deployment.pod_annotations vs values.podAnnotations etc.)?

**Comment by jmazzitelli** [comment]
How I tested:

1. `make build-helm-charts`
2. `helm template --show-only templates/deployment.yaml --set podLabels.testLabel=testValue kiali-operator _output/charts/kiali-operator-1.64.0-SNAPSHOT.tgz`
3. Confirm the `testLabel: testValue` label is in the pod template:

```yaml
spec:
...
  template:
    metadata:
      name: kiali-operator
      namespace: default
      labels:
...
        testLabel: testValue
```

**Review by jmazzitelli — APPROVED** [review]

**Pull request merged**